### PR TITLE
Liquidity mining bindings fix

### PR DESCRIPTION
**Pull request by josefleventon**
https://trello.com/c/XpHBajGR/127-liquidity-mining-revision

### CODE REVIEW

**Comment by Wgil** [comment]
@josefleventon did you test this manually? Can you post here the results please?

**Comment by josefleventon** [comment]
```json
{
  "data": {
    "pools": [
      {
        "pool_id": 1,
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "715000000"
            },
            "weight": "10737418240",
            "usd_value": "714928500"
          },
          {
            "token": {
              "denom": "ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2",
              "amount": "80000000"
            },
            "weight": "10737418240",
            "usd_value": "0"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "0",
        "tvl": "0",
        "rewards": "0"
      },
      {
        "pool_id": 2,
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "97572041044"
            },
            "weight": "10737418240",
            "usd_value": "97562283839"
          },
          {
            "token": {
              "denom": "ibc/E2D2F6ADCC68AA3384B2F5DFACCA437923D137C14E86FB8A10207CF3BED0C8D4",
              "amount": "5319393298"
            },
            "weight": "10737418240",
            "usd_value": "59097031014"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "0.757426472586794706",
        "tvl": "156665.22514844695857823",
        "rewards": "0"
      },
      {
        "pool_id": 3,
        "assets": [
          {
            "token": {
              "denom": "ibc/2180E84E20F5679FCC760D8C165B60F42065DEF7F46A72B447CFF1B7DC6C0A65",
              "amount": "166570245367"
            },
            "weight": "107374182400",
            "usd_value": "166553588342"
          },
          {
            "token": {
              "denom": "uelys",
              "amount": "26769283232"
            },
            "weight": "107374182400",
            "usd_value": "166553588342"
          }
        ],
        "pool_ratio": "50 : 50",
        "rewards_apr": "0",
        "borrow_apr": "0",
        "leverage_lp": "0",
        "perpetual": "1",
        "tvl": "333107.1766849266",
        "rewards": "0"
      }
    ]
  }
}

File: bindings/src/types.rs [hunk 1]
```diff
@@ -357,6 +357,7 @@ pub struct ValidatorDetail {
 pub struct PoolAsset {
     pub token: Coin,
     pub weight: Uint128,
+    pub usd_value: Option<Uint128>,
```
Comment (Wgil) [inline bindings/src/types.rs:360]: I believe this should be `Option<Decimal>` instead. Could you please check how it's being done in other queries?

File: bindings/src/querier.rs [hunk 2]
```diff
@@ -461,7 +461,35 @@ impl<'a> ElysQuerier<'a> {
...
+                            .assets
+                            .iter()
+                            .map(|asset| {
+                                let price = self
+                                    .get_asset_price(asset.token.denom.clone())
+                                    .unwrap_or(Decimal::from_str("0").unwrap());
+                                PoolAsset {
+                                    token: asset.token.clone(),
+                                    weight: asset.weight,
+                                    usd_value: Some(asset.token.amount * price),
```
Comment (Wgil) [inline bindings/src/querier.rs:480]: Also, is this an actual USD value? Can you also verify that as well please? I have seen in other places that conversions go first through ASSET->USDC and then USDC->USD.

**Review by Wgil — CHANGES_REQUESTED** [review]
Just requesting for a couple checks.

File: bindings/src/querier.rs (hunk 2 above)
Comment (josefleventon) [reply bindings/src/querier.rs:480]: https://github.com/elys-network/bindings/blob/78d9f2c5cf333ceceb916af9dba32acce0f9b8ee/bindings/src/querier.rs#L545-L547

Comment (josefleventon) [reply bindings/src/querier.rs]: It should be correct, yes.

Comment (cosmic-vagabond) [reply bindings/src/querier.rs]: price calc correct

**Pull request merged**